src/context.cpp

```diff
@@ -616,12 +639,44 @@ namespace librealsense
             if( option_id == RS2_OPTION_COUNT )
                 throw librealsense::invalid_value_exception( to_string() << "Option " << option->get_name() << " type not found" );
 
-            auto opt = std::make_shared< rs2_dds_option >( option );
+            auto opt = std::make_shared< rs2_dds_option >( option,
+                [&]( const std::string & name, float value ) { set_option( name, value ); },
+                [&]( const std::string & name ) -> float { return query_option( name ); } );
             register_option( option_id, opt );
         }
 
-        //float get_option( rs2_option option ) const override;
-        //void set_option( rs2_option option, float value ) const override;
+        void set_option( const std::string & name, float value ) const
+        {
+            // Sensor is setting the option for all supporting streams (with same value)
+            for( auto & stream : _streams )
+            {
+                for( auto & dds_opt : stream.second->options() )
+                {
+                    if( dds_opt->get_name().compare( name ) == 0 )
+                    {
+                        _dev->set_option_value( dds_opt, value );
+                        break;
+                    }
+                }
+            }
+        }
+
+        float query_option( const std::string & name ) const
+        {
+            for( auto & stream : _streams )
+            {
+                for( auto & dds_opt : stream.second->options() )
+                {
+                    if( dds_opt->get_name().compare( name ) == 0 )
+                    {
+                        //Assumes value is same for all relevant streams in the sensor, values are always set together
```

Comment:
Oh wait, we have streams per sensor?
I thought it is flat on DDS

Reply:
Yes, see previous comment. This is in LibRS sensor context, dds sends to stream